### DISCUSSION TITLE
Alternative Faster Method

### DISCUSSION THREAD

**Discussion by ChanseyIsTheBest** [StevensND/ghidra-port-mods-guide#2] [Ideas]
It's great that you found my Ghidra guide useful and you've made a great guide on how to convert mods. Some people on Discord have informed me of another method to update .ips mods and it's a generally a more straightforward process and doesn't require Ghidra or the long time analyzing. This simpler method may also fail with some games I've heard but requires less technical ability.

1. Dump the main of the game for the update/base game the mod is for and the update of the game you want to update it for
2. Convert the .ips over into a cheat using the python script here: https://github.com/ChanseyIsTheBest/NX-60FPS-RES-GFX-Cheats/blob/main/ips%2Bpchtxt2cheat.py
3. Use Interactive-ASM-Cheat-Updater here https://github.com/zzpong/Interactive-ASM-Cheats-Updater put the old and new main and the converted cheat into it
4. Follow this guide in reverse to convert the cheat into an .pchtxt https://gbatemp.net/threads/how-to-convert-ips-or-pchtxt-into-cheat.626182/
5. Convert the .pchtxt into .ips using IPSwitch https://github.com/3096/ipswitch

**Comment by StevensND** [comment]
Thanks for your contribution!

I decided to move your post from Issues to Discussions since it wasn't really an issue.

I will take a look at it later on my free time. It looks interesting.

My guide is mainly focused on emulation (Yuzu/Ryujinx) due to people on Yuzu/Ryujinx subreddits but I found your posts on Gbatemp useful (not only the one about cheats but also the one about [making 60fps mod](https://gbatemp.net/threads/how-to-make-60fps-ips-patch-for-nintendo-switch-game-ghidra-tutorial.625675/) even if it doesn't work on all games). This has helped me to make mods for Atelier and Master Detective Archives: RAIN Code. Also your Python scripts save a lot of work so they are appreciated :)

I don't know many sites where mods exist in the .ips format so I will use the one you are using for [Sonic Frontiers](https://gbatemp.net/threads/sonic-frontiers-1-1-1-60fps-exefs-patch.625595/). Do you know of any sources of .ips mods for me to try other than this game?

Since my mother tongue is not English I understand that in the first step you mean that I should get 2 mains being the first: the update for which the mod was developed and the second the update I want to use (in this case the latest for Sonic would be 1.3.0). I understand that this would also be the case for step 3 being old in this case the update 1.1.1 and new the update 1.3.0.

If this is so then I think I would have no problem trying your method.

Thanks again for everything!

**Comment by ChanseyIsTheBest** [comment]
All good moving the suggestion to here. I honestly forgot discussions existed lol

The python scripts aren't mine they are a friends' haha 

Sources of .ips patches to practice on include:
https://github.com/masagrator/NXGraphicsPatches
https://github.com/theboy181/switch-ptchtxt-mods
Clicking on the 'Rom Hack' filter in Switch Discussions on GBATemp

_'Since my mother tongue is not English I understand that in the first step you mean that I should get 2 mains being the first: the update for which the mod was developed and the second the update I want to use (in this case the latest for Sonic would be 1.3.0). I understand that this would also be the case for step 3 being old in this case the update 1.1.1 and new the update 1.3.0.'_

Yeah you've got this correct.

**Comment by StevensND** [comment]
> All good moving the suggestion to here. I honestly forgot discussions existed lol
> 
> The python scripts aren't mine they are a friends' haha
> 
> Sources of .ips patches to practice on include: https://github.com/masagrator/NXGraphicsPatches https://github.com/theboy181/switch-ptchtxt-mods Clicking on the 'Rom Hack' filter in Switch Discussions on GBATemp
> 

It was not configured at first haha. By default when you create a repository (at least as I do) you don't have that option. I added it later.

Oh. I thought they were yours haha. I was also modifying them to do the opposite: from cheat to .pchtxt but there are some cheats starting with 580F0000 for example that would not allow to do this. From what I understand you would need to find a new pointer for them unfortunately like someone mentioned [here ](https://www.reddit.com/r/128bitbay/comments/1587i9n/comment/jtq8qzm/?utm_source=share&utm_medium=web2x&context=3) and I don't know if Yuzu + GDB + IDA Pro/Ghidra can do that. I don't have much experience/knowledge using GDB + IDA Pro btw. 

You can see an example of Yuzu + GDB + Ghidra [here](https://youtu.be/_lzU1HAasjo). This is for make an Aspect Ratio Mod for TOTK.

Ok. I will test some of them and see what I get. Thanks for everything again :)

**Comment by StevensND** [comment]
Hello again. I can confirm that for Sonic Frontiers the method worked. However I have a doubt. The code for 1.3.0 that exists in GBATemp adds a new line that the code for 1.1.1 in .ips format didn't have (I think). Would you know tell me that makes that new line?

This is the new line:

`04000000 00268E58 1E200C00`

And the entire cheat is [this ](https://gbatemp.net/threads/sonic-frontiers-1-1-1-60fps-exefs-patch.625595/post-10197799)

I got the same except `04000000 00268E58 1E200C00` line

Below is the mod in .pchtxt format without that line. Game is working at 60fps but I don't know if it's missing something.

![Captura](https://i.imgur.com/fq08yFX.png)

**EDIT:** This info it's just for TOTK but I want you to know it anyway. Interactive-ASM-Cheat-Updater help me with this cheat:

```
[♯ 2. Inf. Health]
080E0000 02B0AFC8 176D734F F9400008
080E0000 02B0AFC0 B9000808 34000048
040E0000 02B0AFBC B9401808
040E0000 00667D04 14928CAE
```

Automatically converts 080X0000 cheats codes into 04. With the method I used before TOTK wouldn't launch directly and would freeze but using Interactive-ASM-Cheat-Updater it now works correctly and I have been able to make the .pchtxt version of that cheat :)

**Reply by ChanseyIsTheBest** [reply]
_Would you know tell me that makes that new line?_
It's possible that ASM cheat updater failed to update a line for whatever reason. That's why manual updating methods like with Ghidra are still important and useful if stuff doesn't work. That's the issue that was raised when some people tried to update Sonic Frontiers actually. I'd check if 60FPS works for menus, cyberspace stages, bosses and cutscenes.

I need to learn how to use GDB myself because it sounds faster then using memory editing software on hardware.

Yeah you can't update cheats unless they start with 04000000 or 04090000 (these are the only two I've heard, it sounds like there are others) which indicate they are main cheats not pointer cheats. Pointer cheats suck to update and are generally easier to update on hardware.

Good to know about the 080X0000 updating thing. I'll look at database to see if any cheats need updating and have that.

**Comment by StevensND** [comment]
[Cheats difference](https://www.diffchecker.com/kbvyCZVp/)

The red side is what I get after using the Python script and the .isp file. The green side is the code for version 1.3.0 from GBATemp. 

Apparently [Interactive-ASM-Cheat-Updater removes line 4](https://i.gyazo.com/5caee07ab7a35b9b54eecb9a6fbe76ab.png) related to `04000000 00271ECC 1E200C00` and `04000000 00268E58 1E200C00` 

I don't know why it does that. However is working fine for the cheat that I mentioned before for TOTK.

Leaving Sonic aside and returning to TOTK I'm curious to know why the following cheat converted to .pchtxt works in both [.pchtxt v1](https://i.gyazo.com/9b801cb0e31ffac6012b7129cd916ceb.png) and [.pchtxt v2 version](https://i.gyazo.com/49e63298dfb0014de068dc2ac2cfc16a.png)

```
[♯ 1. Invincible (Ignore Hits)]
080E0000 02B0AFF8 D65F03C0 6B09011F
080E0000 02B0AFF0 2A0903E8 34000042
080E0000 02B0AFE8 B940EEC2 35000083
080E0000 02B0AFE0 B9406043 F9403842
080E0000 02B0AFD8 F9400442 F94007E2
080E0000 02B0AFD0 37300121 B94032C1
040E0000 007FD9A0 948C358C
```

As you can see in the screenshots the order of the 2nd column in v2 is altered and is not the same as in v1. However both versions work although perhaps V2 is the one that works better even though for some reason the mod doesn't work properly: When starting game it's possible that I receive damage but if I reload the game the mod already works properly. I guess I need to do more tests.

The cheat is [supposed to ignore hits](https://streamable.com/ayuof3) but not damage received by falls, explosions, fire or electricity although I don't know exactly how it works.

_**EDIT:** I have discovered that the problem is the cheat. When you start the game for the first time you receive damage and then you stop receiving it. And if you load the same game a second time the cheat works correctly._

For the v1 version [I used this method ](https://i.gyazo.com/3db66704d4a9d52df5a31192e40cb6f8.png) but it seems that Interactive-ASM-Cheat-Updater alters the order as I told you before. V2 was made using Interactive-ASM-Cheat-Updater

This is really weird to me cause [this cheat ](https://github.com/bad1dea/NXCheats/blob/main/The%20Legend%20of%20Zelda%20Tears%20of%20the%20Kingdom/0100F2C0115B6000%20v327680%20for%20Yuzu/%E2%99%AF%202.%20Inf.%20Health/cheats/6F32C68DD3BC7D77.txt) doesn't work using the same method for V1. Game never launch. However using Interactive-ASM-Cheat-Updater + doing reverse it works without problems. Same for [this cheat ](https://github.com/bad1dea/NXCheats/blob/main/The%20Legend%20of%20Zelda%20Tears%20of%20the%20Kingdom/0100F2C0115B6000%20v327680%20for%20Yuzu/%E2%99%AF15.%20No%20Stamina%20Cost%20For%20Weapon%20Charge/cheats/6F32C68DD3BC7D77.txt) so again ... I'm curious.

If you know what this is due to it would help me a lot since the Reddit comment doesn't give me much information but I more or less understood how it worked. 

Finally regarding GDB ... I'll do another post.

**Comment by StevensND** [comment]
Regarding GDB: Maybe you can take a look at this [Rentry](https://rentry.org/gdbghidratuto) 

It's a recompilation of all the information I took out to make the video I passed you about GDB before. It has a lot of interesting information. 

I think you can use GDB with real hardware too but I'm not pretty sure. Read [this ](https://gist.github.com/jam1garner/c9ba6c0cff150f1a2480d0c18ff05e33)
